### PR TITLE
Disable `TransformSubnetTx` post-Etna

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Per [ACP-77](https://github.com/avalanche-foundation/ACPs/blob/main/ACPs/77-reinventing-subnets/README.md):

> After this ACP is activated, Elastic Subnets will be disabled. `TransformSubnetTx` will not be accepted post-activation.

## How this works

Marks `TransformSubnetTx` as invalid after EUpgrade is activated

## How this was tested

Added UT

### CODE REVIEW

File: vms/platformvm/txs/executor/standard_tx_executor.go [hunk 1]
```diff
@@ -436,6 +437,10 @@ func (e *StandardTxExecutor) TransformSubnetTx(tx *txs.TransformSubnetTx) error
 		return err
 	}
 
+	if e.Config.UpgradeConfig.IsEActivated(currentTimestamp) {
```
[the remark this answers is not in the document]
Comment (dhrubabasu) [reply vms/platformvm/txs/executor/standard_tx_executor.go]: it's a better pattern to fail fast- I've moved it up good call

**Comment by dhrubabasu** [comment]
> The e2e failure appears legit. Maybe worth enabling e2e runs against different rulesets (i.e. #2666) in advance of the E upgrade?

I've deleted the test for now... I think we'll need to start thinking about testing different rulesets for the E Upgrade

Not super concerned with this specific one because this functionality isn't used by anybody on Mainnet but if there's a good way to restore the e2e I would definitely prefer that

File: vms/platformvm/txs/executor/standard_tx_executor_test.go [hunk 2]
```diff
@@ -1552,6 +1552,49 @@ func TestDurangoMemoField(t *testing.T) {
 	}
 }
 
+// Verifies that [TransformSubnetTx] is disabled post-E
+func TestEUpgradeDisabledTransactions(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (abi87) [reply vms/platformvm/txs/executor/standard_tx_executor_test.go:1556]: I wonder if this should be more general? Like test, at each relevant fork, what txs are allowed/forbidden?

[the remark this answers is not in the document]
Comment (dhrubabasu) [reply vms/platformvm/txs/executor/standard_tx_executor_test.go:1556]: I left it open-ended so we could extend it easily with other transactions we may wish to disable in `E` (matches `TestDurangoDisabledTransactions`)

File: tests/e2e/p/elastic_subnets.go [hunk 3]
```diff
@@ -34,6 +35,19 @@ var _ = e2e.DescribePChain("[Permissionless Subnets]", func() {
...
 
+			infoClient := info.NewClient(nodeURI.URI)
+
+			tc.By("get upgrade config")
+			upgrades, err := infoClient.Upgrades(tc.DefaultContext())
+			require.NoError(err)
+
+			now := time.Now()
+			if upgrades.IsEtnaActivated(now) {
+				tc.Outf("{{green}}Etna is activated{{/}}: %s (now) >= %s (EtnaTime)\n", now, upgrades.EtnaTime)
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply tests/e2e/p/elastic_subnets.go:46]: We will be adding tests that skip if etna _isn't_ activated. So seems like something to be clean if possible. `ginkgo.Skip` is definitely the right move 👍

File: vms/platformvm/txs/executor/standard_tx_executor_test.go [hunk 4]
```diff
@@ -1597,6 +1597,49 @@ func TestDurangoMemoField(t *testing.T) {
...
 
+// Verifies that [TransformSubnetTx] is disabled post-Etna
+func TestEtnaDisabledTransactions(t *testing.T) {
+	type test struct {
+		name        string
+		buildTx     func(*environment) *txs.Tx
+		expectedErr error
+	}
+
+	tests := []test{
```
[the remark this answers is not in the document]
Comment (dhrubabasu) [reply vms/platformvm/txs/executor/standard_tx_executor_test.go]: Removed the scaffolding, I think this one is the only one

File: tests/e2e/p/elastic_subnets.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhrubabasu) [reply tests/e2e/p/elastic_subnets.go]: Modified to ginkgo.Skip. This will probably be the only one that is not expected to pass pre-Etna.

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**